src/body/tee_body.rs

```diff
@@ -75,11 +74,13 @@ impl Body for TeeBody {
             if let Some(mut next) = backlog.pop_front() {
                 match io.as_mut().poll_write(context, &next) {
                     Poll::Pending => {
-                        backlog.push_front(next);
                         // Placeholder for tracing...
-                        if polled_frame {
-                            return Poll::Pending;
-                        }
+                        //
+                        // Something along the lines of:
+                        // tracing::info!("TeeBody: io is not yet ready for writes.");
+                        //
+                        backlog.push_front(next);
+                        return Poll::Pending;
```

Comment:
This introduces back pressure in the event that the AsyncWrite is not ready for writes. We're only storing 1 data frame in the backlog while we wait for the socket to become available because the additional capacity in the VecDeque is in place exclusively for partial writes as seen in the match arm for `Poll::Ready(Ok(len))`.

Reply:
It's possible but very unlikely that a slower implementation of AsyncWrite could cause the VecDeque in IoState::Writeable to allocate. Rather than trying to prevent this from happening, I think it's best that we add tracing or placeholders for tracing so we know if / when we end up needing to allocate additional capacity for the backlog. This metric will likely be useful in determining the size we end up choosing for `MAX_FRAME_LEN` in ResponseBody. We can slowly move towards `64kb` frames, if we see that TeeBody requires an additional allocation when paired with a TcpStream or an io stream such as stderr, we can adjust accordingly.